### PR TITLE
CI Build Test 

### PR DESCRIPTION
**Pull request by cbduggap**
Changes to support the capability to Configure
TemporaryRamSize by Bootloader

### CODE REVIEW

**Comment by ChaselChiu** [comment]
Looks good to me! Thanks for working on this!

File: IntelFsp2Pkg/FspSecCore/SecFsp.c [hunk 1]
```diff
@@ -68,11 +68,11 @@ SecGetPlatformData (
   FspPlatformData->CodeRegionBase      = 0;
   FspPlatformData->CodeRegionSize      = 0;
 
-  //
-  // Pointer to the size field
-  //
-  TopOfCar = PcdGet32 (PcdTemporaryRamBase) + PcdGet32 (PcdTemporaryRamSize);
-  StackPtr = (UINT32 *)(TopOfCar - sizeof (UINT32));
+  if ((TopOfCar == 0) || (TopOfCar == 0xFFFFFFFF)) {
```
Comment (tedkuo1) [inline IntelFsp2Pkg/FspSecCore/SecFsp.c:71]: If above 4G is not supported for temp ram base, please consider changing it as below.
if ((TopOfCar == 0) || (TopOfCar >= MAX_UINT32)) {

File: IntelFsp2Pkg/FspSecCore/Ia32/FspApiEntryT.nasm [hunk 2]
```diff
@@ -506,7 +485,7 @@ ASM_PFX(EstablishStackFsp):
   cmp       byte [edx + LoadMicrocodeParamsFsp22.FspUpdHeaderRevision], 2
   jb        Fsp20UpdHeader2
   cmp       byte [esp + LoadMicrocodeParamsFsp22.FsptArchRevision], 2
-  je        Fsp24UpdHeader2
+  jae        Fsp24UpdHeader2
```
Comment (tedkuo1) [inline IntelFsp2Pkg/FspSecCore/Ia32/FspApiEntryT.nasm:488]: Please align Fsp24UpdHeader2 with above line.

File: IntelFsp2Pkg/FspSecCore/Ia32/FspApiEntryT.nasm [hunk 3]
```diff
@@ -405,7 +383,7 @@ CheckAddress:
    cmp   byte [esp + LoadMicrocodeParamsFsp22.FspUpdHeaderRevision], 2
    jb     Fsp20UpdHeader1
    cmp    byte [esp + LoadMicrocodeParamsFsp22.FsptArchRevision], 2
-   je     Fsp24UpdHeader1;
+   jae     Fsp24UpdHeader1;
```
Comment (tedkuo1) [inline IntelFsp2Pkg/FspSecCore/Ia32/FspApiEntryT.nasm:386]: Please align Fsp24UpdHeader1 with above line.

File: IntelFsp2Pkg/FspSecCore/Ia32/FspApiEntryT.nasm [hunk 4]
```diff
@@ -267,7 +245,7 @@ ASM_PFX(LoadMicrocodeDefault):
    cmp    byte [esp + LoadMicrocodeParamsFsp22.FspUpdHeaderRevision], 2
    jb     Fsp20UpdHeader
    cmp    byte [esp + LoadMicrocodeParamsFsp22.FsptArchRevision], 2
-   je     Fsp24UpdHeader
+   jae     Fsp24UpdHeader
```
Comment (tedkuo1) [inline IntelFsp2Pkg/FspSecCore/Ia32/FspApiEntryT.nasm:248]: Please align Fsp24UpdHeader with above line.

File: IntelFsp2Pkg/FspSecCore/X64/FspApiEntryT.nasm [hunk 5]
```diff
@@ -163,7 +165,7 @@ ASM_PFX(LoadMicrocodeDefault):
    cmp    byte [rsp + LoadMicrocodeParamsFsp24.FspUpdHeaderRevision], 2
    jb     ParamError
    cmp    byte [rsp + LoadMicrocodeParamsFsp24.FsptArchRevision], 2
-   jne    ParamError
+   jb    ParamError
```
Comment (tedkuo1) [inline IntelFsp2Pkg/FspSecCore/X64/FspApiEntryT.nasm:168]: Please align ParamError with above line.

File: IntelFsp2Pkg/FspSecCore/X64/FspApiEntryT.nasm [hunk 6]
```diff
@@ -273,7 +275,7 @@ CheckAddress:
    cmp   byte [rsp + LoadMicrocodeParamsFsp24.FspUpdHeaderRevision], 2
    jb    ParamError
    cmp   byte [rsp + LoadMicrocodeParamsFsp24.FsptArchRevision], 2
-   jne   ParamError
+   jb   ParamError
```
Comment (tedkuo1) [inline IntelFsp2Pkg/FspSecCore/X64/FspApiEntryT.nasm:278]: Please align ParamError with above line.

File: IntelFsp2Pkg/FspSecCore/X64/FspApiEntryT.nasm [hunk 7]
```diff
@@ -349,7 +358,7 @@ ASM_PFX(EstablishStackFsp):
   cmp       byte [rdx + LoadMicrocodeParamsFsp24.FspUpdHeaderRevision], 2
   jb        ParamError1
   cmp       byte [rdx + LoadMicrocodeParamsFsp24.FsptArchRevision], 2
-  je        Fsp24UpdHeader
+  jnb        Fsp24UpdHeader
```
Comment (tedkuo1) [inline IntelFsp2Pkg/FspSecCore/X64/FspApiEntryT.nasm:361]: Please align Fsp24UpdHeader with above line.

File: IntelFsp2Pkg/FspSecCore/X64/FspApiEntryT.nasm [hunk 8]
```diff
@@ -463,6 +477,47 @@ ParamValid:
   or        rax, rdx
   SAVE_TS   rax
 
+  mov       rdx, ASM_PFX(PcdGet32 (PcdTemporaryRamSize))
+  mov       edx, DWORD[rdx]
```
Comment (tedkuo1) [inline IntelFsp2Pkg/FspSecCore/X64/FspApiEntryT.nasm:481]: Please insert one space right before [.

File: IntelFsp2Pkg/FspSecCore/X64/FspApiEntryT.nasm [hunk 9]
```diff
@@ -463,6 +477,47 @@ ParamValid:
...
+  ;
+  cmp       ebx, edx
+  je        UseTemporaryRamSizePcd
+  mov       rax, 08000000000000002h      ; RETURN_INVALID_PARAMETER
+  jmp       TempRamInitExit
+ConsumeInputConfiguration:
+  ;
+  ; Read Fsp Arch2 revision
+  cmp       byte [ecx + LoadMicrocodeParamsFsp24.FsptArchRevision], 3
+  jb       UseTemporaryRamSizePcd
```
Comment (tedkuo1) [inline IntelFsp2Pkg/FspSecCore/X64/FspApiEntryT.nasm:510]: Please align UseTemporaryRamSizePcd with above line.

**Pull request closed without merging**